File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.swift
import Foundation

func gcd(_ a:Int, _ b:Int) -> Int {
    if a % b == 0 {
        return b
    }
    return gcd(b, a%b)
}

func solution(_ numer1:Int, _ denom1:Int, _ numer2:Int, _ denom2:Int) -> [Int] {
    var maxNumer = (numer1*denom2) + (numer2*denom1)
    var maxDenom = denom1 * denom2
    var gcdValue = gcd(maxNumer, maxDenom)
    var result = [maxNumer/gcdValue, maxDenom/gcdValue]
    return result
}

File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int maxDenum = denom1 * denom2;
        int maxNumer = (numer1 * denom2) + (numer2 * denom1);
//         기약분수 만들기
        for (int i = maxNumer-1; i > 1; i--) {
            if (maxNumer % i == 0 && maxDenum % i == 0 ) {
                maxNumer /= i;
                maxDenum /= i;
            }
        }
        int[] answer = {maxNumer, maxDenum};
        return answer;
    }
}